src/wad/tex.rs

```diff
@@ -248,11 +261,10 @@ impl TextureDirectory {
         (tex, bound_map)
     }
 
-    pub fn build_flat_atlas<'a, T: Iterator<Item = &'a WadName>>(
-            &'a self, names_iter: T) -> (OpaqueImage, BoundsLookup) {
-        let names = ordered_atlas_entries(
-            &self.animated_flats, |n| self.flat(&n),
-            names_iter);
+    pub fn build_flat_atlas<'a, T: Iterator<Item = &'a WadName>>(&'a self,
```

Comment:
Another place where a where clause probably makes more sense, could you please change?